napari/utils/processes.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import uuid
+from enum import auto
+from typing import Optional
+
+from napari.utils.misc import StringEnum
+
+
+class ProcessStatus(StringEnum):
+    BUSY = auto()
+    IDLE = auto()
```

Comment:
What does `IDLE` mean here? Something like pending? Might be useful to have also something like `Done`.